test/e2e/aws_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	regopapi "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1alpha1"
+	"github.com/openshift/cluster-image-registry-operator/pkg/clusterconfig"
+
+	clientcore "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+func TestAWS(t *testing.T) {
+
+	kubeconfig, err := clusterconfig.GetConfig()
+	if err != nil {
+		t.Errorf("unable to obtain cluster configuration: %#v", err)
+	}
+
+	client, err := clientcore.NewForConfig(kubeconfig)
+	if err != nil {
+		t.Errorf("unable to get client: %#v", err)
+	}
+
+	cfg, err := clusterconfig.Get()
+	if err != nil {
+		t.Errorf("unable to get cluster configuration: %#v", err)
+	}
+
+	// If the storage type is not S3, skip this test.
+	if regopapi.ImageRegistryStorageType(cfg.Storage.Type) != regopapi.StorageTypeS3 {
```

Comment:
It's better to change the type of the Type variable.

Reply:
updated